src/util/string.h

```diff
@@ -75,9 +85,12 @@ inline std::string MakeUnorderedList(const std::vector<std::string>& items)
 /**
  * Check if a string does not contain any embedded NUL (\0) characters
  */
-[[nodiscard]] inline bool ValidAsCString(const std::string& str) noexcept
+[[nodiscard]] inline bool ValidAsCString(std::string_view str) noexcept
```

Comment:
I think the name `ValidAsCString` is now a bit dangerous. I'd assume that it returns `true` when the argument is 0 terminated at the end (which was the case for the `const std::string& str` argument), but with a `std::string_view` argument this might not be the case.

So now even when `ValidAsCString(str)` returns `true`, it is not allowed to use e.g. `strlen(str.data())`.

I would rename the method to something like `ContainsNoNUL` to make it more clear what's checked, or keep the `const std::string&` argument for now

Reply:
According to the documentation, it only checks if a string has embedded `\0` characters. That std::string is zero-terminated internally is an implementation detail, the caller is still responsible for passing `c_str()` (and never `.data()`) when passing it to a function that takes a C string.

Reply:
well, maybe it's just me, but from the interface `ValidAsCString(std::string_view str)` I'd guess that it returns `true` when it actually finds a `\0` somewhere, but it's exactly the opposite :slightly_smiling_face:

Reply:
I have no objection to adding a scripted-diff to rename `ValidAsCString` to `ContainsNoNUL`. Either in this pull or as a follow-up.

Reply:
@sipa

Reply:
@martinus See https://github.com/bitcoin/bitcoin/pull/25001